From b8fb42f015e01a16a2efa4c8d479c079355bc6e2 Mon Sep 17 00:00:00 2001
From: xarantolus <xarantolus@protonmail.com>
Date: Mon, 20 Jun 2022 22:52:54 +0200
Subject: [PATCH 1/2] Add wasm compiler version

---
 Makefile | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 8716957..aca0620 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,5 @@
 CC_win=x86_64-w64-mingw32-gcc
+CC_wasm=emcc
 
 PLATFORM_MACRO=
 ifeq ($(OS),Windows_NT)     # is Windows_NT on XP, 2000, 7, Vista, 10...
@@ -22,22 +23,26 @@ DESTDIR=/usr/local/bin
 # Files to compile
 FILES=compiler/memeasm.c compiler/compiler.c compiler/logger/log.c compiler/parser/parser.c compiler/parser/fileParser.c compiler/parser/functionParser.c compiler/analyser/analysisHelper.c compiler/analyser/parameters.c compiler/analyser/functions.c compiler/analyser/jumpMarkers.c compiler/analyser/comparisons.c compiler/analyser/randomCommands.c compiler/analyser/analyser.c compiler/translator/translator.c
 
-.PHONY: all clean debug uninstall install windows
+.PHONY: all clean debug uninstall install windows wasm
 
 # Standard compilation
 all:
 	$(CC) -o memeasm $(FILES) $(CFLAGS)
 
+wasm: wasm/memeasm.js
+wasm/memeasm.js:
+	$(CC_wasm) -o $@ -s WASM=1 -s EXPORT_ES6=1 -s MODULARIZE=1 -s EXPORTED_RUNTIME_METHODS="['FS', 'callMain']" -s 'EXPORT_NAME="runMemeAssemblyCompiler"' $(FILES) $(CFLAGS)
+
 # Compilation with debugging-flags
 debug:
 	$(CC) -o memeasm $(FILES) $(CFLAGS) $(CFLAGS_DEBUG)
 
 # Remove the compiled executable from this directory
-clean: 
+clean:
 	$(RM) memeasm
 
 # Removes "memeasm" from DESTDIR
-uninstall: 
+uninstall:
 	$(RM) $(DESTDIR)/memeasm
 
 # Compiles an executable and stores it in DESTDIR
@@ -47,4 +52,4 @@ install:
 
 # For building a windows executable under Linux
 windows:
-	$(CC_win) -o memeasm.exe $(FILES) $(CFLAGS)
\ No newline at end of file
+	$(CC_win) -o memeasm.exe $(FILES) $(CFLAGS)
-- 
2.25.1

