From d4984015f716f85673c7aebb2b07f27bd1bfda40 Mon Sep 17 00:00:00 2001
From: xarantolus <xarantolus@protonmail.com>
Date: Wed, 28 Sep 2022 11:29:33 +0200
Subject: [PATCH] Fix optind

---
 compiler/memeasm.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/compiler/memeasm.c b/compiler/memeasm.c
index e6c9eea..71e6902 100644
--- a/compiler/memeasm.c
+++ b/compiler/memeasm.c
@@ -81,6 +81,10 @@ int main(int argc, char* argv[]) {
     int opt;
     int option_index = 0;
 
+    // On normal systems one wouldn't have to reset optind, but the WebAssembly version
+    // requires global variables to be reset; else option parsing wouldn't work when calling main again
+    optind = 1;
+
     while ((opt = getopt_long_only(argc, argv, "o:hO::digS", long_options, &option_index)) != -1) {
         switch (opt) {
             case 'h':
-- 
2.34.1

